src/library/browse/browsefeature.cpp

```diff
@@ -18,6 +18,9 @@
 #include "widget/wlibrarysidebar.h"
 #include "widget/wlibrarytextbrowser.h"
 
+bool s_browseLimitChildItemsEnabled;
+int s_browseLimitChildItemsNumber;
```

Comment:
This is not what I meant. I'm sorry I've been unclear with my previous suggestion. I was rather suggesting you make these two local variables and initialize them just before the loop. With `ConfigKey` globals I was suggesting to just create them as constants (and also employ `QStringLiteral`s):

```suggestion
const ConfigKey kChildItemLimitKey = ConfigKey(QStringLiteral("[Library]"), QStringLiteral("BrowseFilesystemLimitChildItemsNumber")));
const ConfigKey kChildItemsLimitEnabledKey = ConfigKey(QStringLiteral("[Library]"), QStringLiteral("BrowseFilesystemLimitChildItemsEnabled"))
```

In general, avoid writeable globals whenever possible, they make it easy to write race conditions and don't work well in an OOP environment.

Reply:
>globals

No problem, I thought it was a little bit odd, but you asked for globals ;-)

>What about enabling this by default?
... will be disappointed by Mixxx ... something like 100, ... "Load more" ... bottleneck ... 200

I can't  find any arguments against your idea. I guess this is how a coffee-project becomes a big dinnerparty project 😄 
-> default 100 -> popup continue/stop with reference to setting. we don't want users to be disappointed.
-> bottleneck= filesystem; chrono= opening/displaying a folder with 100 items takes the same time as opening a folder with 3200 items: +/- 59secs

OwKee

Reply:
> -> bottleneck= filesystem; chrono= opening/displaying a folder with 100 items takes the same time as opening a folder with 3200 items: +/- 59secs

Okay, just to be sure: you measured that with the Browse feature, right?

Reply:
Yes, to simulate the issue: Browse via wired LAN to NAS. (local drive is faster)
Why?

Reply:
Just wanted to make sure there are no false assumptions due comparing Browse with Crates or something, that's all.